tests/sentry/snuba/test_discover.py

```diff
@@ -0,0 +1,390 @@
+
+from sentry.api.event_search import InvalidSearchQuery
+from sentry.snuba import discover
+from sentry.testutils import TestCase, SnubaTestCase
+from sentry.testutils.helpers.datetime import iso_format, before_now
+from sentry.utils.snuba import Dataset
+
+from mock import patch
+import pytest
+
+
+class QueryIntegrationTest(SnubaTestCase, TestCase):
+    def setUp(self):
+        super(QueryIntegrationTest, self).setUp()
+        self.environment = self.create_environment(self.project, name="prod")
+        self.release = self.create_release(self.project, version="first-release")
+
+        self.store_event(
+            data={
+                "message": "oh no",
+                "release": "first-release",
+                "environment": "prod",
+                "platform": "python",
+                "user": {"id": "99", "email": "bruce@example.com", "username": "brucew"},
+                "timestamp": iso_format(before_now(minutes=1)),
+            },
+            project_id=self.project.id,
+        )
+
+    def test_field_aliasing_in_selected_columns(self):
+        result = discover.query(
+            selected_columns=["project.id", "user.email", "release"],
+            query="",
+            params={"project_id": [self.project.id]},
+        )
+        data = result["data"]
+        assert len(data) == 1
+        assert data[0]["project.id"] == self.project.id
+        assert data[0]["user.email"] == "bruce@example.com"
+        assert data[0]["release"] == "first-release"
+
+    def test_field_aliasing_in_aggregate_functions_and_groupby(self):
+        result = discover.query(
+            selected_columns=["project.id", "count_unique(user.email)"],
+            query="",
+            params={"project_id": [self.project.id]},
+        )
+        data = result["data"]
+        assert len(data) == 1
+        assert data[0]["project.id"] == self.project.id
+        assert data[0]["count_unique_user_email"] == 1
+
+    def test_field_aliasing_in_conditions(self):
+        result = discover.query(
+            selected_columns=["project.id", "user.email"],
+            query="user.email:bruce@example.com",
+            params={"project_id": [self.project.id]},
+        )
+        data = result["data"]
+        assert len(data) == 1
+        assert data[0]["project.id"] == self.project.id
+        assert data[0]["user.email"] == "bruce@example.com"
+
+    def test_release_condition(self):
+        result = discover.query(
+            selected_columns=["id", "message"],
+            query="release:{}".format(self.create_release(self.project).version),
+            params={"project_id": [self.project.id]},
+        )
+        assert len(result["data"]) == 0
+
+        result = discover.query(
+            selected_columns=["id", "message"],
+            query="release:{}".format(self.release.version),
+            params={"project_id": [self.project.id]},
+        )
+        assert len(result["data"]) == 1
```

Comment:
Can we assert on the responses, and that they have the right keys that were transformed in them?